lib/puppet/resource/type.rb

```diff
@@ -360,9 +360,6 @@ def set_resource_parameters(resource, scope)
     resource.add_parameters_from_consume
     inject_external_parameters(resource, scope)
 
-    resource_hash = {}
```

Comment:
Two comments about that.
How did we learn that this is client side only?
Isn't there a fair chance that if we add this to some methods, we might create a false impression that it's added to all methods where it makes sense?

I'm not at all against making the distinction. I just think it in order to help it must be done throughout. Adding it to some methods might actually add to the confusion.